### DISCUSSION TITLE
IP Routes cannot be persistently SET or REMOVED

### DISCUSSION THREAD

**Discussion by Zarox666** [home-assistant/operating-system#2827] [Q&A] [answered]
### Describe the issue you are experiencing

Independant if IP Routes are removed via **_ip route del_** or via **_route del_** they reapear after restart.
Analog any routes set via **_ip route add_** or **_via route add_** disapear after restart.

### What operating system image do you use?

ova (for Virtual Machines)

### What version of Home Assistant Operating System is installed?

6.1.56

### Did you upgrade the Operating System.

Yes

### Steps to reproduce the issue

1. Set a IP route
2. Restart
3. Check if IP Route is still there

### Anything in the Supervisor logs that might be useful for us?

```txt
nope
```

### Anything in the Host logs that might be useful for us?

```txt
nope
```

### System information

<html>
<body>
<h2 id="title" class="mdc-dialog__title" style="-webkit-font-smoothing: antialiased; font-family: var(--mdc-typography-headline6-font-family,var(--mdc-typography-font-family,Roboto,sans-serif)); font-size: var(--mdc-typography-headline6-font-size,1.25rem); line-height: var(--mdc-typography-headline6-line-height,2rem); font-weight: var(--mdc-typography-headline6-font-weight,500); letter-spacing: var(--mdc-typography-headline6-letter-spacing,.0125em); text-decoration: var(--mdc-typography-headline6-text-decoration,inherit); text-transform: var(--mdc-typography-headline6-text-transform,inherit); display: block; position: relative; flex-shrink: 0; box-sizing: border-box; margin: 0px 0px 1px; padding: 24px 24px 0px; color: var(--mdc-dialog-heading-ink-color,rgba(0,0,0,.87));"><div class="header_title" style="margin-right: 32px; margin-inline-end: 32px; margin-inline-start: initial; direction: var(--direction);">System information</div><ha-icon-button dialogaction="close" class="header_button" style="display: inline-block; outline: 0px; position: absolute; right: 16px; top: 14px; text-decoration: none; color: inherit; inset-inline-start: initial; inset-inline-end: 16px; direction: var(--direction);"><mwc-icon-button title="Close" style="display: inline-block; outline: 0px; --mdc-ripple-color: currentcolor; -webkit-tap-highlight-color: transparent; vertical-align: top; --mdc-theme-on-primary: currentColor; --mdc-theme-text-disabled-on-light: var(--disabled-text-color);"><button class="mdc-icon-button mdc-icon-button--display-flex" aria-label="Close" style="font-size: 24px; width: var(--mdc-icon-button-size,48px); height: var(--mdc-icon-button-size,48px); padding: calc((var(--mdc-icon-button-size,48px) - var(--mdc-icon-size,24px))/ 2); display: inline-flex; position: relative; box-sizing: border-box; border: none; outline: 0px; background-color: transparent; fill: currentcolor; color: inherit; text-decoration: none; cursor: pointer; user-select: none; z-index: 0; overflow: visible; align-items: center; justify-content: center; vertical-align: top;"><span><slot><ha-svg-icon style="display: block; align-items: center; justify-content: center; position: relative; vertical-align: middle; fill: currentcolor; width: var(--mdc-icon-size,24px); height: var(--mdc-icon-size,24px);"><svg preserveAspectRatio="xMidYMid meet" focusable="false" role="img" aria-hidden="true" viewBox="0 0 24 24"><g><path d="M19,6.41L17.59,5L12,10.59L6.41,5L5,6.41L10.59,12L5,17.59L6.41,19L12,13.41L17.59,19L19,17.59L13.41,12L19,6.41Z"></path></g></svg></ha-svg-icon></slot></span></button></mwc-icon-button></ha-icon-button></h2><div id="content" class="mdc-dialog__content" style="-webkit-font-smoothing: antialiased; font-family: var(--mdc-typography-body1-font-family,var(--mdc-typography-font-family,Roboto,sans-serif)); font-size: var(--mdc-typography-body1-font-size,1rem); line-height: var(--mdc-typography-body1-line-height,1.5rem); font-weight: var(--mdc-typography-body1-font-weight,400); letter-spacing: var(--mdc-typography-body1-letter-spacing,.03125em); text-decoration: var(--mdc-typography-body1-text-decoration,inherit); text-transform: var(--mdc-typography-body1-text-transform,inherit); flex-grow: 1; box-sizing: border-box; margin: 0px; overflow: auto; color: var(--mdc-dialog-content-ink-color,rgba(0,0,0,.6)); padding: var(--dialog-content-padding,24px); position: var(--dialog-content-position,relative);"><slot id="contentSlot" style="margin-top: 0px; margin-bottom: 0px;"><div><div>

Version | core-2023.10.3
-- | --
Installation Type | Home Assistant OS
Development | false
Supervisor | true
Docker | true
User | root
Virtual Environment | false
Python Version | 3.11.5
Operating System Family | Linux
Operating System Version | 6.1.56
CPU Architecture | x86_64
Timezone | Europe/Berlin
Configuration Directory | /config

</div><div><h3>Core metrics</h3><ha-metric><ha-settings-row style="display: flex; padding: 0px; align-content: normal; align-self: auto; align-items: center; height: 54px; width: 537px;"><div class="prefix-wrap" style="display: var(--settings-row-prefix-display);"><slot name="prefix"></slot><div class="body" two-line="" style="padding: 8px 16px 8px 0px; overflow: hidden; display: var(--layout-vertical_-_display); flex-direction: var(--layout-vertical_-_flex-direction); justify-content: var(--layout-center-justified_-_justify-content); flex: 1 1 0%; min-height: calc(var(--paper-item-body-two-line-min-height,72px) - 16px);"><slot name="heading" style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"><span slot="heading">Processor usage</span></slot><div class="secondary" style="overflow: hidden; text-overflow: ellipsis; white-space: normal; display: block; padding-top: 4px; font-family: var(--mdc-typography-body2-font-family,var(--mdc-typography-font-family,Roboto,sans-serif)); -webkit-font-smoothing: antialiased; font-size: var(--mdc-typography-body2-font-size,.875rem); font-weight: var(--mdc-typography-body2-font-weight,400); line-height: normal; color: var(--secondary-text-color);"><slot name="description"><div slot="description" title="" style="white-space: normal; color: var(--secondary-text-color); display: flex; justify-content: space-between;"><span class="value" style="width: 48px; padding-right: 4px; flex-shrink: 0;">0.1<span> </span>%</span><ha-bar class="  " style="--ha-bar-primary-color: var(
          --metric-bar-ok-color,
          var(--success-color)
        );"><svg><g><rect></rect><rect width="0.1%"></rect></g></svg></ha-bar></div></slot></div></div></div><div class="content" style="display: var(--settings-row-content-display,flex); justify-content: flex-end; flex: 1 1 0%; padding: 16px 0px;"><slot></slot></div></ha-settings-row></ha-metric><ha-metric><ha-settings-row style="display: flex; padding: 0px; align-content: normal; align-self: auto; align-items: center; height: 54px; width: 537px;"><div class="prefix-wrap" style="display: var(--settings-row-prefix-display);"><slot name="prefix"></slot><div class="body" two-line="" style="padding: 8px 16px 8px 0px; overflow: hidden; display: var(--layout-vertical_-_display); flex-direction: var(--layout-vertical_-_flex-direction); justify-content: var(--layout-center-justified_-_justify-content); flex: 1 1 0%; min-height: calc(var(--paper-item-body-two-line-min-height,72px) - 16px);"><slot name="heading" style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"><span slot="heading">Memory usage</span></slot><div class="secondary" style="overflow: hidden; text-overflow: ellipsis; white-space: normal; display: block; padding-top: 4px; font-family: var(--mdc-typography-body2-font-family,var(--mdc-typography-font-family,Roboto,sans-serif)); -webkit-font-smoothing: antialiased; font-size: var(--mdc-typography-body2-font-size,.875rem); font-weight: var(--mdc-typography-body2-font-weight,400); line-height: normal; color: var(--secondary-text-color);"><slot name="description"><div slot="description" title="" style="white-space: normal; color: var(--secondary-text-color); display: flex; justify-content: space-between;"><span class="value" style="width: 48px; padding-right: 4px; flex-shrink: 0;">2.1<span> </span>%</span><ha-bar class="  " style="--ha-bar-primary-color: var(
          --metric-bar-ok-color,
          var(--success-color)
        );"><svg><g><rect></rect><rect width="2.1%"></rect></g></svg></ha-bar></div></slot></div></div></div><div class="content" style="display: var(--settings-row-content-display,flex); justify-content: flex-end; flex: 1 1 0%; padding: 16px 0px;"><slot></slot></div></ha-settings-row></ha-metric><h3>Supervisor metrics</h3><ha-metric><ha-settings-row style="display: flex; padding: 0px; align-content: normal; align-self: auto; align-items: center; height: 54px; width: 537px;"><div class="prefix-wrap" style="display: var(--settings-row-prefix-display);"><slot name="prefix"></slot><div class="body" two-line="" style="padding: 8px 16px 8px 0px; overflow: hidden; display: var(--layout-vertical_-_display); flex-direction: var(--layout-vertical_-_flex-direction); justify-content: var(--layout-center-justified_-_justify-content); flex: 1 1 0%; min-height: calc(var(--paper-item-body-two-line-min-height,72px) - 16px);"><slot name="heading" style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"><span slot="heading">Processor usage</span></slot><div class="secondary" style="overflow: hidden; text-overflow: ellipsis; white-space: normal; display: block; padding-top: 4px; font-family: var(--mdc-typography-body2-font-family,var(--mdc-typography-font-family,Roboto,sans-serif)); -webkit-font-smoothing: antialiased; font-size: var(--mdc-typography-body2-font-size,.875rem); font-weight: var(--mdc-typography-body2-font-weight,400); line-height: normal; color: var(--secondary-text-color);"><slot name="description"><div slot="description" title="" style="white-space: normal; color: var(--secondary-text-color); display: flex; justify-content: space-between;"><span class="value" style="width: 48px; padding-right: 4px; flex-shrink: 0;">0<span> </span>%</span><ha-bar class="  " style="--ha-bar-primary-color: var(
          --metric-bar-ok-color,
          var(--success-color)
        );"><svg><g><rect></rect><rect width="0%"></rect></g></svg></ha-bar></div></slot></div></div></div><div class="content" style="display: var(--settings-row-content-display,flex); justify-content: flex-end; flex: 1 1 0%; padding: 16px 0px;"><slot></slot></div></ha-settings-row></ha-metric><ha-metric><ha-settings-row style="display: flex; padding: 0px; align-content: normal; align-self: auto; align-items: center; height: 54px; width: 537px;"><div class="prefix-wrap" style="display: var(--settings-row-prefix-display); color: rgb(225, 225, 225); font-family: Roboto, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: 0.4375px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(28, 28, 28); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><slot name="prefix"></slot><div class="body" two-line="" style="padding: 8px 16px 8px 0px; overflow: hidden; display: var(--layout-vertical_-_display); flex-direction: var(--layout-vertical_-_flex-direction); justify-content: var(--layout-center-justified_-_justify-content); flex: 1 1 0%; min-height: calc(var(--paper-item-body-two-line-min-height,72px) - 16px);"><slot name="heading" style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"><span slot="heading">Memory usage</span></slot><div class="secondary" style="overflow: hidden; text-overflow: ellipsis; white-space: normal; display: block; padding-top: 4px; font-family: var(--mdc-typography-body2-font-family,var(--mdc-typography-font-family,Roboto,sans-serif)); -webkit-font-smoothing: antialiased; font-size: var(--mdc-typography-body2-font-size,.875rem); font-weight: var(--mdc-typography-body2-font-weight,400); line-height: normal; color: var(--secondary-text-color);"><slot name="description"><div slot="description" title="" style="white-space: normal; color: var(--secondary-text-color); display: flex; justify-content: space-between;"><span class="value" style="width: 48px; padding-right: 4px; flex-shrink: 0;">0.7<span> </span>%</span><ha-bar class="  " style="--ha-bar-primary-color: var(
          --metric-bar-ok-color,
          var(--success-color)
        );"><svg><g><rect></rect><rect width="0.7%"></rect></g></svg></ha-bar></div></slot></div></div></div><br class="Apple-interchange-newline">
</body>
</html>

### Additional information

When I started using HA about 1 year ago this worked, however I cannot say what HA Version that was.
I have succesfully added routes in the config that are still there today.
However due to changes in the Network Infrastructure these routes need to be deleted now, thus I discovered that no modification (neither deleting nor adding) to the routing table survives a reboot.
Also see: https://community.home-assistant.io/t/declare-a-static-route-to-reach-other-lan/311442/8

**Comment by agners** [comment]
The `ip route` command changes the Linux (kernel) routing table directly. This routing table is (and always was) volatile. This means any `ip route` command, on HAOS as well as other Linux distributions, is not persistent.

Typically, routes get added automatically (e.g. as learned by the DHCP server).

The `ha network` command currently doesn't support static routes. You'd need to use the underlying NetworkManager directly. You can check if you've added static routes this way previously using

```
nmcli con show "Supervisor enp1s0"
```

(replace `enp1s0` with the primary network interface name in your installation)


To clear any static routes, something along this lines should work:

```
nmcli con modify "Supervisor enp1s0" ipv4.routes ""
```

**Reply by Zarox666** [reply]
as a Windows guy I find it hillarious that you cannot set persistent routes by the route command.
Anyway... yes nmcli showed the routes I try to remove.
However when I type the modify command it fails:
![image](https://github.com/home-assistant/operating-system/assets/115449639/34001e03-1987-473a-abed-f42b0056929d)

Interesting as it even suggests ipv4.routes "" for the autocomplete. Not sure why its going on about ipv4.dns-data....

**Reply by agners** [reply]
I've seen that previously, but don't remember what the problem was exactly :see_no_evil: 

Can you clear that first?

```
nmcli con modify "Supervisor enp1s0" ipv4.dns-data ""
```

**Comment by Zarox666** [comment]
> I've seen that previously, but don't remember what the problem was exactly 🙈

lol, jeah I know that too well myself :-)
Thanks for the help by the way.

When I do an nmcli con show it does not show any field called ipv4.dns-data.
thus when I try to empty the value it says
![image](https://github.com/home-assistant/operating-system/assets/115449639/f2f77b1a-536b-4481-8a0f-e6c4d5e807af)

**Reply by Zarox666** [reply]
Well well searching the internet....
look what I found: 

https://github.com/hassio-addons/addon-ssh/issues/583

So far I tried running it through the SSH Addon... let me try on the VM directly.

**Reply by Zarox666** [reply]
ok that one does not even know nmcli
![image](https://github.com/home-assistant/operating-system/assets/115449639/845e7717-0c6a-4827-8225-95f1ecdf1a5f)

**Reply by agners** [reply]
You need to use `login` to get to the OS shell.

**Reply by Zarox666** [reply, marked as answer]
thanks!
There it works.

So to sum up:
If you want to configure (add or remove) static routes you need to 
a) do it through the nmcli con modify "Your connection name" ipv4.routes Comand
b) you have to execute this comand on the VM itself after login in to it and not through the SSH Terminal addon.

Let me restart to see if the routes stay gone.

**Reply by Zarox666** [reply]
It worked :-) Thanks!
